airflow/contrib/operators/gcf_function_deploy_operator.py

```diff
@@ -0,0 +1,266 @@
+    dict(name="runtime", regexp=r'^nodejs6$|^nodejs8$|^python37$', optional=True),
+    dict(name="timeout", regexp=r'^\d+(\.\d{1,9})?s$', optional=True),
+    dict(name="availableMemoryMb", custom_validation=_validate_available_memory_in_mb,
+         optional=True),
+    dict(name="labels", custom_validation=_validate_labels_dict, optional=True),
+    dict(name="environmentVariables", custom_validation=_validate_env_vars_dict,
+         optional=True),
+    dict(name="network",
+         regexp=r'^[^\s]+$',
+         optional=True),
+    dict(name="maxInstances", optional=True, custom_validation=_validate_max_instances),
+
+    dict(name="source_code", type="union", fields=[
+        dict(name="sourceArchiveUrl", regexp=r'^gs://[^\s]+$'),
+        dict(name="sourceRepositoryUrl", regexp=r'^gs://[^\s]+$', api='v1beta2'),
+        dict(name="sourceRepository", type="dict", fields=[
+            dict(name="url",
+                 regexp=r'^[^\s]+$')
+        ]),
+        dict(name="sourceUploadUrl", regexp=r'^https://[^\s]+$')
+    ]),
+
+    dict(name="trigger", type="union", fields=[
+        dict(name="httpsTrigger", type="dict", fields=[
+            # This dict should be empty at input (url is added at output)
+        ]),
+        dict(name="eventTrigger", type="dict", fields=[
+            dict(name="eventType", regexp=r'^[^\s]+$'),
+            dict(name="resource", regexp=r'^[^/]+$'),
+            dict(name="service", regexp=r'^[^\s]+$', optional=True),
+            dict(name="failurePolicy", type="dict", optional=True, fields=[
+                dict(name="retry", type="dict", optional=True)
+            ])
+        ])
+    ]),
+]
+
+
+class GCFFunctionDeployOperator(BaseOperator):
+    """
+    Create a function in Google Cloud Functions.
+
+    :param project_id: Project ID that the operator works on
+    :type project_id: str
+    :param region: Region where the operator operates on
+    :type region: str
+    :param body: Body of the cloud function definition. The body must be CloudFunction
+        dictionary as described in:
+   https://cloud.google.com/functions/docs/reference/rest/v1/projects.locations.functions
+        (note that different API versions require different
+        variants of the CloudFunction dictionary)
+    :type body: dict or google.cloud.functions.v1.CloudFunction
+    :param gcp_conn_id: The connection ID to use connecting to Google Cloud Platform.
+    :type gcp_conn_id: str
+    :param api_version: Version of the API used (for example v1).
+    :type api_version: str
+    :param zip_path: Path to zip file containing source code of the function. I
+    f it is set, then sourceUploadUrl should not be specified in the body (or it should
+    be empty), then the zip file will be uploaded using upload URL generated
+    via generateUploadUrl from cloud functions API
+    :type zip_path: str
+    """
+
+    @apply_defaults
+    def __init__(self,
+                 project_id,
+                 region,
```

Comment:
Change region to location_id according to the doc.

Reply:
Are you sure about location_id ? I reviewed the docs (SOW) and it looks it should be 'location' rather than 'location_id'. I changed it to `location` in  38d0b6d99ddc55e6e281625ec3a072a95aceb927